content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,312 @@
+
+Don't worry if you don't include this information as in that case docker will automatically assign a name to the image. However specifying a name is advisable as it will allow you to identify your images more easily.
+
+You can view a list of all your docker images in your Docker Desktop dashboard or by typing `docker images` in the command line.
+
+{{% /tip %}}
+
+{{% codeblock %}}
+```bash
+# Build a Docker image and assign a name to it
+$ docker build -t <your-image-name> .
+```
+{{% /codeblock %}}
+
+### Step 3: Docker compose
+
+Docker compose is a tool within the Docker ecosytem whose main functionality is to coordinate your containers and provide them with the services required to run smoothly. These services include actions such as communication between containers or starting up containers which require aditional instructions to do so. Although this can sound a bit confusing if you are new to Docker (You can learn more about Docker compose [here](https://docs.docker.com/compose/)), the key takeaway is that it will assist and simplify the task of replicating your environment by providing some extra information to Docker about how to do so. Additionally it is worth noting that we could perform these actions manually employing the `docker run` command, however in this way the process not as straightforward.
+
+In the same way that you needed a dockerfile to provide Docker with the necessary instructions to build an image, a docker-compose file is required to tell docker how to provide these services to a (series of) container(s).
+
+{{% tip %}}
+
+Docker compose files must be YAML files, so make sure that your docker compose file is denoted as such with the extension ".yml" 
+
+{{% /tip %}}
+
+Now it is time to upload your Docker-compose in the same way you did it with your dockerfile, and then and excute the following code in the directory where your docker-compose file is located:
+
+{{% codeblock %}}
+```bash
+# Build the docker image
+$ docker compose up 
+```
+{{% /codeblock %}}
+
+{{% warning %}}
+
+If you receive an error message after executing the codeblock above, run `sudo chmod 666 /var/run/docker.sock` and try it again.
+
+{{% /warning %}}
+
+ If the execution of docker-compose (and all its services) was succesful that means that your envorinment is up and running in jupyter and you should see something in your command line that resembles what is shown in the image below.
+
+<p align = "center">
+<img src = "../img/jupyter_on.png" width="750">
+<figcaption> Typical log of a Jupyter instance, signaling you that the replication of the environment was completed. </figcaption>
+</p>
+
+{{% tip %}}
+
+You can also specify which services do you want to run within a docker-compose file by executing the command shown above followed by the name of the service desired: `docker compose up <service-to-be-run>` .
+
+{{% /tip %}}
+
+### Extra! - Connect the Docker container within your instance to a Google Cloud Storage bucket 

```

Comment:
Mayeb make this a sub title since now the extra is just as important as the normal steps. So i would change the ### to ####

Reply:
That makes sense, done!